File: python/palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        newStr = ""
        for c in s:
            if c.isalnum():
                newStr += c.lower()
        return newStr == newStr[::-1]


# Test cases
if __name__ == "__main__":
    solution = Solution()
    print(solution.isPalindrome("A man, a plan, a canal: Panama"))  # True
    print(solution.isPalindrome("race a car"))  # False
    print(solution.isPalindrome(" "))  # True
    print(solution.isPalindrome("bob"))

File: js/palindrome.js
class Solution {
    isPalindrome(s) {
        // Remove non-alphanumeric characters and convert to lowercase
        s = s.replace(/[^a-z0-9]/gi, '').toLowerCase();

        // Check if the string is equal to its reverse
        return s === s.split('').reverse().join('');
    }
}

// Example usage
const solution = new Solution();
console.log(solution.isPalindrome("A man, a plan, a canal: Panama")); // true
console.log(solution.isPalindrome("race a car")); // false
console.log(solution.isPalindrome(" ")); // true
console.log(solution.isPalindrome("bob")); // true
console.log(solution.isPalindrome("Madam")); // true